File: frontend/src/app/features/tic-tac-toe/models/gameStateDto.ts
import { GameResult } from "../../../shared/enums/gameResult";

export interface GameStateDto {
    /** Unikalny identyfikator gry (UUID w Javie) */
    gameId: string;
    /** Plansza 3x3. Typ komórki to 'X', 'O' lub null (puste). */
    board: (string | null)[][];
    /** Następny gracz, który ma wykonać ruch ('X' lub 'O') */
    nextPlayer: string;
    /** Rezultat gry. Null, jeśli gra nadal trwa. */
    result: GameResult | null;
    /** Wiadomość statusowa lub komunikat dla gracza */
    message: string;

    winningSide: string;

    wonCoordinates: string;
}

File: backend/src/main/java/az/project/VocabularyBuilder/tictactoe/dto/GameStateDto.java
package az.project.VocabularyBuilder.tictactoe.dto;

import az.project.VocabularyBuilder.tictactoe.domain.GameResult;

import java.util.UUID;

public record GameStateDto(
        UUID gameId,
        String[][] board,
        String nextPlayer,
        GameResult result, // Null, jeśli gra trwa
        String message,
        String winningSide,
        String wonCoordinates
) {
}
